Src/AmrCore/AMReX_ErrorList.H

```diff
@@ -372,6 +376,86 @@ private:
 
 std::ostream& operator << (std::ostream& os, const ErrorList& elst);
 
+  struct AMRErrorTagInfo
+  {
+    int            m_max_level;
+    amrex::Real    m_min_time;
+    amrex::Real    m_max_time;
+    amrex::RealBox m_realbox;
+
+    AMRErrorTagInfo () noexcept : m_max_level(-1), m_min_time(-1), m_max_time(-1), m_realbox(amrex::RealBox()) {}
+
+    AMRErrorTagInfo& SetMaxLevel (int max_level) noexcept {
+      m_max_level = max_level;
+      return *this;
+    }
+    AMRErrorTagInfo& SetMinTime (amrex::Real min_time) noexcept {
+      m_min_time = min_time;
+      return *this;
+    }
+    AMRErrorTagInfo& SetMaxTime (amrex::Real max_time) noexcept {
+      m_max_time = max_time;
+      return *this;
+    }
+    AMRErrorTagInfo& SetRealBox (const amrex::RealBox& realbox) noexcept {
+      m_realbox = realbox;
+      return *this;
+    }
+  };
+
+  class AMRErrorTag
+  {
+  public:
+
+    enum TEST {GRAD=0, LESS, GREATER, VORT, BOX, USER};
+
+    struct UserFunc
+    {
+      virtual void operator() (const amrex::Box&                       bx,
+                               amrex::Array4<const amrex::Real> const& dat,
+                               amrex::Array4<char> const&              tag,
+                               amrex::Real                             time,
+                               int                                     level,
+                               int                                     tagval,
+                               int                                     clearval) = 0;
+    };
+
+    AMRErrorTag (const AMRErrorTagInfo& info = AMRErrorTagInfo()) noexcept
```

Comment:
Should we add `explicit`?

Reply:
I don't think so, as that would require an info object, correct?  If the defaults are ok, this is the one type of case I'm ok with default arguments.... if I'm wrong about ``explicit`` let me know

Reply:
explicit just means if there is a function that takes `AMRErrorTag` as argument, one cannot just pass `AMRErrorTagInfo` and let compiler to implicit convert it.

Reply:
https://godbolt.org/z/6hh89h

Reply:
You can still have the default argument to the constructor.